### PR TITLE
Fix memory leaks in SWIG wrappers.

### PR DESCRIPTION
**Pull request by kkg4theweb**
Fixes memory leaks in the SWIG wrappers for meep::make_output_directory and the automatic freeing of material_type_list

### CODE REVIEW

File: python/meep.i [hunk 1]
```diff
@@ -1405,6 +1405,7 @@ void _get_gradient(PyObject *grad, PyObject *fields_a, PyObject *fields_f, PyObj
     if ($1.num_items != 0) {
         for (int i = 0; i < $1.num_items; i++) {
             material_free($1.items[i]);
+            delete $1.items[i];
```
Comment (stevengj) [inline python/meep.i:1408]: Maybe it would be better to put this in `material_free`?

Comment (kkg4theweb) [reply python/meep.i:1408]: We could - but then it might break the code of anybody using the C++ api directly.
I dont know how much we worry about that.

Comment (stevengj) [reply python/meep.i:1408]: I'm not so worried about that.

Comment (stevengj) [reply python/meep.i]: I pushed a patch to move `delete` into `material_free`.   It looks like there was another memory leak in `set_default_material` that this fixes.

Comment (kkg4theweb) [reply python/meep.i]: Thanks for making the change. I tested this internally and everything looks good.
Please go ahead and merge this change.

**Pull request merged**